### DISCUSSION TITLE
(issue) $ symfony console make:migration

### DISCUSSION THREAD

**Discussion by ylyco** [symfony/symfony#44377] [Symfony Book: The Fast Track]
Hi all,

I'm learning Symfony from the french paper book version " En route pour Symfony 5".

1)
When I try to migrate there is my error (see chapter 8.7) :

$ symfony console make:migration

In ExceptionConverter.php line 73:
                                                                                                      
  An exception occurred in the driver: SQLSTATE[08006] [7] server closed the connection unexpectedly  
        This probably means the server terminated abnormally                                                
        before or while processing the request.

2) 
I also have this issue too : 

$ symfony run psql
psql: erreur : la connexion au serveur a été coupée de façon inattendue
        Le serveur s'est peut-être arrêté anormalement avant ou durant le
        traitement de la requête.
                                        
  command "psql" failed: exit status 2

3) 
postgresql seems fine, because I have : 

$ docker-compose ps

Name                      Command              State                     Ports                   
guestbook_database_1   docker-entrypoint.sh postgres   Up      0.0.0.0:49153->5432/tcp,:::49153->5432/tcp

4)
Note : Any suggestion is welcome.

Thanks in advance.

**Comment by ylyco** [comment]
another information of interest : 

$ symfony var:export
PGDATABASE=main SYMFONY_TUNNEL_ENV= SYMFONY_DOCKER_ENV=1 DATABASE_URL=postgres://main:main@127.0.0.1:49153/main?sslmode=disable&charset=utf8 DATABASE_SERVER=postgres://127.0.0.1:49153 DATABASE_DRIVER=postgres DATABASE_PORT=49153 DATABASE_USERNAME=main DATABASE_DATABASE=main PGHOST=127.0.0.1 DATABASE_HOST=127.0.0.1 DATABASE_USER=main PGPASSWORD=main PGUSER=main PGPORT=49153 DATABASE_PASSWORD=main DATABASE_NAME=main SYMFONY_TUNNEL=[regis@localhost guestbook]$

**Comment by michaoj** [comment]
Did you try to set the DATABASE_URL var in the .env file as mentionned in the book on chapter 8.3? If yes, what' is the value of this variable?

**Comment by ylyco** [comment]
@michaoj  Thank you for your reply

content .env : 

DATABASE_URL="postgresql://symfony:ChangeMe@127.0.0.1:5432/app?serverVersion=13&charset=utf8"


content .env.local : 

PGHOST=127.0.0.1
PGPORT=49153
PGDATABASE=main
PGUSER=main
PGPASSWORD=main


What do you think about that ?
Thanks :)

**Comment by ylyco** [comment]
I changed to : 

DATABASE_URL="postgresql://main:main@127.0.0.1:49153/app?serverVersion=13&charset=utf8"

But the issue still remains...

**Comment by ylyco** [comment]
DATABASE_URL="postgresql://127.0.0.1:5432/db?serverVersion=13&charset=utf8"

This doesn't work out neither ...

**Comment by ylyco** [comment]
One thing I don't understand : 

I change .env to : 
DATABASE_URL="postgresql://127.0.0.1:5432/app?serverVersion=13&charset=utf8"

But the command do not take in count the modification even if I restart docker service : 
$ symfony var:export
PGHOST=127.0.0.1 PGPASSWORD=main PGUSER=main DATABASE_NAME=main DATABASE_PORT=49153 **DATABASE_URL=postgres://main:main@127.0.0.1:49153/main?sslmode=disable&charset=utf8**

It still show another DATABASE_URL than the one in .env file ...

**Reply by michaoj** [reply]
Do you have a .env.local file which override the .env file ?

By the way can you connect to your postgresql server by using the command line or a client (dbeaver, datagrip, other,...)

**Reply by ylyco** [reply]
> Do you have a .env.local file which override the .env file ?
> 
> By the way can you connect to your postgresql server by using the command line or a client (dbeaver, datagrip, other,...)

This is the .env.local content : 

PGHOST=127.0.0.1
PGPORT=49153
PGDATABASE=main
PGUSER=main
PGPASSWORD=main

And this is the connection to postgresql server command : 

[regis@localhost guestbook]$ systemctl start docker
[regis@localhost guestbook]$ docker-compose up -d
Starting guestbook_database_1 ... done
[regis@localhost guestbook]$ docker-compose ps
        Name                      Command              State                     Ports                   
---------------------------------------------------------------------------------------------------------
guestbook_database_1   docker-entrypoint.sh postgres   Up      0.0.0.0:49153->5432/tcp,:::49153->5432/tcp
[regis@localhost guestbook]$ docker exec -it guestbook_database_1 psql -U main -W main
Password: 
psql (13.5)
Type "help" for help.

main=# \l
                             List of databases
   Name    | Owner | Encoding |  Collate   |   Ctype    | Access privileges 
-----------+-------+----------+------------+------------+-------------------
 main      | main  | UTF8     | en_US.utf8 | en_US.utf8 | 
 postgres  | main  | UTF8     | en_US.utf8 | en_US.utf8 | 
 template0 | main  | UTF8     | en_US.utf8 | en_US.utf8 | =c/main          +
           |       |          |            |            | main=CTc/main
 template1 | main  | UTF8     | en_US.utf8 | en_US.utf8 | =c/main          +
           |       |          |            |            | main=CTc/main
(4 rows)

main=# \q
[regis@localhost guestbook]$ 

Thank you :)

**Reply by michaoj** [reply]
The .env.local override the .env file, thus you need to modify it if you want to see the changes reflected

**Comment by ylyco** [comment]
I delete every lines in .env.local, and restart docker : the DATABASE_URL still the same one : 
 
```
[regis@localhost guestbook]$ symfony var:export
SYMFONY_DOCKER_ENV= SYMFONY_TUNNEL= SYMFONY_TUNNEL_ENV=
[regis@localhost guestbook]$ ps -ef | grep docker
root        7866       1  0 10:09 ?        00:00:00 /usr/sbin/dockerd -g /var/cache/docker -H unix://var/run/docker.sock -H tcp://127.0.0.1:2375
root        7892    7866  0 10:09 ?        00:00:00 containerd --config /var/run/docker/containerd/containerd.toml --log-level info
regis       8260    7506  0 10:10 pts/0    00:00:00 grep --color docker
[regis@localhost guestbook]$ docker-compose ps
        Name                      Command               State                       Ports                   
------------------------------------------------------------------------------------------------------------
guestbook_database_1   docker-entrypoint.sh postgres   Exit 255   0.0.0.0:49153->5432/tcp,:::49153->5432/tcp
[regis@localhost guestbook]$ docker-compose up -d
Starting guestbook_database_1 ... done
[regis@localhost guestbook]$ docker-compose ps
        Name                      Command              State                     Ports                   
---------------------------------------------------------------------------------------------------------
guestbook_database_1   docker-entrypoint.sh postgres   Up      0.0.0.0:49153->5432/tcp,:::49153->5432/tcp
[regis@localhost guestbook]$ symfony var:export
DATABASE_USERNAME=main DATABASE_PORT=49153 DATABASE_HOST=127.0.0.1 PGUSER=main PGHOST=127.0.0.1 PGPORT=49153 SYMFONY_TUNNEL= SYMFONY_TUNNEL_ENV= PGDATABASE=main DATABASE_USER=main DATABASE_DRIVER=postgres DATABASE_DATABASE=main **DATABASE_URL=postgres://main:main@127.0.0.1:49153/main?sslmode=disable&charset=utf8** SYMFONY_DOCKER_ENV=1 DATABASE_PASSWORD=main PGPASSWORD=main DATABASE_SERVER=postgres://127.0.0.1:49153 DATABASE_NAME=main
[regis@localhost guestbook]$ symfony run psql
psql: erreur : la connexion au serveur a été coupée de façon inattendue
        Le serveur s'est peut-être arrêté anormalement avant ou durant le
        traitement de la requête.
                                        
  command "psql" failed: exit status 2  
```

**Comment by michaoj** [comment]
Do you have a psql client on your machine ?

If so, could you try this :

`psql -U main -h 127.0.0.1 -p 49153 -d main`

I just want to check you can connect in another context than the symfony one

**Comment by ylyco** [comment]
Your command doesn't work for me : 

[regis@localhost guestbook]$ psql -U main -h 127.0.0.1 -p 49153 main
psql: erreur : la connexion au serveur a été coupée de façon inattendue
        Le serveur s'est peut-être arrêté anormalement avant ou durant le
        traitement de la requête.

But this one does : 

[regis@localhost guestbook]$ docker exec -it guestbook_database_1 psql -U main -W main
Password: 
psql (13.5)
Type "help" for help.

main=# 


For your consideration : 

[regis@localhost ~]$ sudo -i -u postgres
[sudo] Mot de passe de regis : 
[postgres@localhost ~]$ ps -ef | grep postgres
70          8598    8579  0 10:10 ?        00:00:00 postgres
70          8809    8598  0 10:10 ?        00:00:00 postgres: checkpointer 
70          8810    8598  0 10:10 ?        00:00:00 postgres: background writer 
70          8811    8598  0 10:10 ?        00:00:00 postgres: walwriter 
70          8812    8598  0 10:10 ?        00:00:00 postgres: autovacuum launcher 
70          8813    8598  0 10:10 ?        00:00:00 postgres: stats collector 
70          8814    8598  0 10:10 ?        00:00:00 postgres: logical replication launcher 
70         23364    8598  0 11:39 ?        00:00:00 postgres: main main [local] idle
root       23917   23855  0 11:41 pts/1    00:00:00 sudo -i -u postgres
postgres   23928   23917  0 11:41 pts/1    00:00:00 -bash
postgres   23991   23928  0 11:42 pts/1    00:00:00 ps -ef
postgres   23992   23928  0 11:42 pts/1    00:00:00 grep --color postgres
[postgres@localhost ~]$ psql 
psql: error: could not connect to server: No such file or directory
        Is the server running locally and accepting
        connections on Unix domain socket "/tmp/.s.PGSQL.5432"?
[postgres@localhost ~]$ su
Password: 
[root@localhost pgsql]$ service postgresql start
Redirecting to /bin/systemctl start postgresql.service
[root@localhost pgsql]$ exit
exit
[postgres@localhost ~]$ psql 
psql (13.5)
Type "help" for help.

postgres=# 



Thank you for your time

**Comment by ylyco** [comment]
Another clue : 

if I empty the content of .env.local, and I comment in .env file the line DATABASE_URL="postgresql://127.0.0.1:5432/db?serverVersion=13&charset=utf8"

I then launch the command " docker-compose up -d "

The system still found the DATABASE_URL value : 

[regis@localhost guestbook]$ symfony var:export
SYMFONY_TUNNEL_ENV= DATABASE_DATABASE=main PGUSER=main PGPASSWORD=main DATABASE_HOST=127.0.0.1 DATABASE_PASSWORD=main SYMFONY_TUNNEL= PGDATABASE=main PGHOST=127.0.0.1 DATABASE_DRIVER=postgres DATABASE_PORT=49153 DATABASE_USER=main DATABASE_NAME=main PGPORT=49153 SYMFONY_DOCKER_ENV=1 **_DATABASE_URL=postgres://main:main@127.0.0.1:49153/main?sslmode=disable&charset=utf8_** DATABASE_SERVER=postgres://127.0.0.1:49153 DATABASE_USERNAME=main
[regis@localhost guestbook]$ 

Does anyone understand where this DATABASE_URL value comes from ?

Thank you for your help

**Comment by ylyco** [comment]
Hi all,

I reached to use psql using : 

$ symfony server:start -d
$ systemctl start docker
$ docker-compose up -d
$ symfony run psql

as a very newbie I wasn't launching the symfony server, so " symfony console make:migration " couldn't work properly.

Problem solved.

Thank you  michaoj for your time
